sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1298,7 +1312,7 @@ class Analyzer(
             s"does not match the required number of columns ($requiredColumns)")
         }
```

Comment:
Why we need to issue the error message in `rewriteSubQuery`? Instead of here? If we do it here, we do not need to pass `e.isInstanceOf[ScalarSubquery]` to `rewriteSubQuery `